core/src/main/java/org/geysermc/geyser/entity/type/TextDisplayEntity.java

```diff
@@ -56,6 +57,11 @@ public void setDisplayNameVisible(BooleanEntityMetadata entityMetadata) {
         // On JE: Hiding the display name still shows the display entity text.
     }
 
+    @Override
+    public void setDisplayName(EntityMetadata<Optional<Component>, ?> entityMetadata) {
+        // Don't allow the display name to be set - messes with our armor stand.
```

Comment:
```suggestion
        // This would usually set EntityDataTypes.NAME, but we are instead using NAME for the text display.
        // On JE: custom name does not override text display.
```